### PR TITLE
add compound query docs

### PR DESCRIPTION
**Pull request by djfritz**

### CODE REVIEW

File: search/search.md
Comment (traetox) [inline search/search.md:118]: lets change this terminology to `inner query` and `main query`

File: search/search.md
Comment (traetox) [inline search/search.md:122]: Can be used by any module that can use a lookup resource, the `regex`, `geoip`, etc.. modules don't know how to use the table based resources.

File: search/search.md
Comment (traetox) [inline search/search.md:150]: the lookup module has a "strict" `-s` flag so that you don't need the `require` module, might be worth being very concise and showing the "fastest" queries for examples.  Verbosity has merits too, so your call.

**Review by traetox — CHANGES_REQUESTED** [review]
some wording changes and a clarification on which modules can use the temporary resources.

File: search/search.md
Comment (djfritz) [reply search/search.md:118]: done

File: search/search.md
Comment (djfritz) [reply search/search.md]: Added a table of supported modules and added more specific language

File: search/search.md
Comment (djfritz) [reply search/search.md]: Changed example and language to just use lookup with -s and -v

**Comment by djfritz** [comment]
@traetox PTAL

**Comment by djfritz** [comment]
@traetox PTAL

**Review by traetox — APPROVED** [review]
LGTM, merge when beta lands

File: search/search.md
Comment (floren) [inline search/search.md:139]: not a valid query, need json module or something

File: search/search.md
Comment (floren) [inline search/search.md:149]: same, make this a valid query

File: search/search.md
Comment (floren) [inline search/search.md:155]: we need to show where SrcIP is coming from

Comment (floren) [reply search/search.md:155]: and DstIP, SrcIPBytes, and DstIPBytes

**Review by floren — CHANGES_REQUESTED** [review]

**Review by floren — APPROVED** [review]

**Pull request merged**
